exareme2/algorithms/exaflow/compute_average.py

```diff
@@ -0,0 +1,56 @@
+from exareme2.algorithms.exaflow.algorithm import Algorithm
+from exareme2.algorithms.exaflow.exaflow_registry import exaflow_udf
+from exareme2.worker_communication import ColumnDataFloat
+from exareme2.worker_communication import ColumnDataStr
+from exareme2.worker_communication import TabularDataResult
+
+ALGORITHM_NAME = "compute_average"
+
+
+class ComputeAverage(Algorithm, algname=ALGORITHM_NAME):
+    def run(self, inputdata, metadata):
+        local_results = self.engine.run_algorithm_step(
+            func="compute_average_local_step",
+            positional_args={"inputdata": inputdata},
+        )
+
+        # Assume inputdata["y"] is now a list of column names.
+        results = {}
+        for column in inputdata["y"]:
+            global_sum = sum(res[column]["sum"] for res in local_results)
+            global_count = sum(res[column]["count"] for res in local_results)
+            average = global_sum / global_count if global_count else float("nan")
+            results[column] = average
+
+        # Prepare result columns
+        variables = list(results.keys())
+        averages = list(results.values())
+
+        return TabularDataResult(
+            title="Average",
+            columns=[
+                ColumnDataStr(name="variable", data=variables),
+                ColumnDataFloat(name="average", data=averages),
+            ],
+        )
+
+
+def compute_local_sum_and_count(columns, data):
+    results = {}
+    for column in columns:
+        col_data = data[column]
+        results[column] = {
+            "sum": float(col_data.sum()),
+            "count": int(col_data.count()),
+        }
+    return results
+
+
+@exaflow_udf(name="compute_average_local_step")
```

Comment:
We don't need a `name` parameter. 

We will use the module/algorithm name and the udf name as key for the registry.

```def make_unique_func_name(func) -> str:
    """Creates a unique function name composed of the function name, an
    underscore and the module's name hashed, encoded in base32 and truncated at
    4 chars."""
    full_module_name = func.__module__
    module_name = full_module_name.split(".")[-1]
    hash_ = get_base32_hash(module_name)
    qualname = func.__qualname__.replace(".", "__")
    return qualname + "_" + hash_.lower()
 ```
 
 Use something like this, hash is not needed.